src/sokoban.cpp

```diff
@@ -218,20 +229,55 @@ bool Sokoban::undo() {
     moves.pop_back();
     history.pop_back();
 
-    _board = history.back();
+    _board = history.back().first;
     locate_player();
 
     return true;
 }
 
+bool Sokoban::rewind() {
+    
+    if (moves.empty()) {
+        return false;
+    }
+    
+    undo();
+
+    // Reset fast_forward if the previous state is landmarked as fast_forward
```

Comment:
We shouldn't leave old code commented out like this. It's in an older commit if we ever need it so I'd suggest removing it entirely.